src/core/router/history/hash.js

```diff
@@ -18,7 +18,15 @@ export class HashHistory extends History {
     const path = window.location.pathname || '';
     const base = this.config.basePath;
 
-    return /^(\/|https?:)/g.test(base) ? base : cleanPath(path + '/' + base);
+    // This handles the case where Docsify is served off an
+    // explicit file path, i.e.`/base/index.html#/blah`. This
+    // prevents the `/index.html` part of the URI from being
+    // remove during routing.
+    // See here: https://github.com/docsifyjs/docsify/pull/1372
+    const basePath = path.endsWith('.html')
```

Comment:
[`String.prototype.endsWith()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith#browser_compatibility) is not supported by IE11 (which Docsify v4 supports). We'll need to use a legacy-compatible method here.

Reply:
Fixed in ac5ed770ff25ee7ebb52bf893d1269b30580549b.